### PR TITLE
DOCS-1178: Add S3 API Compatibility List

### LINKED ISSUE #1178: S3 API Support Matrix
**Summary**

We have an internal ask to formally document the list of supported S3 APIs.

**Goals**

- [ ] Walk the list of S3 APIs present in https://github.com/minio/minio/blob/master/cmd/api-router.go
- [ ] Document the list at a high level (tabular format)
- [ ] Link (where possible) to the equivalent S3 API

**Non-Goals**

- Documenting the actual API request/response
- Documenting any MinIO Admin APIs
- Documenting the usage of the API

### PR DESCRIPTION
**Pull request by ravindk89**
Closes #1178 

Staged: http://192.241.195.202:9000/staging/DOCS-1178/linux/reference/s3-api-compatibility.html

Summary:

This covers the S3 APIs which we directly support - reference minio/minio/cmd/api-router.go

Any APIs which are marked as 'dummy' internally are considered 'unsupported'
Admin APIs are out of scope here
MinIO API extensions are out of scope as well

We are not doing any detailed reference - just pointing out to S3

### CODE REVIEW

File: source/reference/s3-api-compatibility.rst [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+- :s3-api:`PutObjectRetention <API_PutObjectRetention.html>`
+- :s3-api:`GetObjectLegalHold <API_GetObjectLegalHold.html>`
+- :s3-api:`PutObjectLegalHold <API_PutObjectLegalHold.html>`
+- :s3-api:`GetObjectLockConfiguration <API_GetObjectLockConfiguration.html>`
+- :s3-api:`PutObjectLockConfiguration <API_PutObjectLockConfiguration.html>`
+
+Multipart Uploads
+~~~~~~~~~~~~~~~~~
+
+:s3-api:`AbortMultipartUpload <API_AbortMultipartUpload.html>`
```
Comment (djwfyi) [inline source/reference/s3-api-compatibility.rst:50]: These need to be list items

File: source/reference/s3-api-compatibility.rst [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+Bucket Policies
+~~~~~~~~~~~~~~~
+
+- :s3-api:`GetBucketPolicy <API_GetBucketPolicy.html>`
+- :s3-api:`GetBucketPolicyStatus <API_GetBucketPolicyStatus.html>`
+- :s3-api:`PutBucketPolicy <API_PutBucketPolicy.html>`
+- :s3-api:`DeleteBucketPolicy <API_DeleteBucketPolicy.html>`
+
+Unsupported API Endpoints
```
Comment (djwfyi) [inline source/reference/s3-api-compatibility.rst:119]:
```suggestion
Unsupported API Bucket Endpoints
```

File: source/reference/s3-api-compatibility.rst [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+:s3-api:`AbortMultipartUpload <API_AbortMultipartUpload.html>`
+:s3-api:`CompleteMultipartUpload <API_CompleteMultipartUpload.html>`
+:s3-api:`CreateMultipartUpload <API_CreateMultipartUpload.html>`
+:s3-api:`ListMultipartUploads <API_ListMultipartUploads.html>`
+:s3-api:`ListParts <API_ListParts.html>`
+:s3-api:`UploadPart <API_UploadPart.html>`
+:s3-api:`UploadPartCopy <API_UploadPartCopy.html>`
+
+Unsupported API Endpoints
```
Comment (djwfyi) [inline source/reference/s3-api-compatibility.rst:58]:
```suggestion
Unsupported API Object Endpoints
```

File: source/reference/s3-api-compatibility.rst [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: none
```
Comment (djwfyi) [inline source/reference/s3-api-compatibility.rst:61]: Sphinx is bulleting this list, but not the one for buckets below. Weird.

Maybe try
```suggestion
.. code-block:: text
```

File: source/reference/s3-api-compatibility.rst [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+S3 API Compatibility
+====================
+
+.. default-domain:: minio
+
+.. contents:: Table of Contents
+   :local:
+   :depth: 2
+
+This page documents all S3 APIs supported by MinIO Object Storage.
```
Comment (djwfyi) [inline source/reference/s3-api-compatibility.rst:11]: I'd drop `all`.
```suggestion
This page documents S3 APIs supported by MinIO Object Storage.
```

**Review by djwfyi — APPROVED** [review]
Couple of suggestions for clarity and a weirdness in rendering between the two unsupported lists.

Otherwise LGTM

**Review by feorlen — APPROVED** [review]
What Daryl said, otherwise 👍🏻

**Review by krisis — APPROVED** [review]

**Pull request merged**